controllers/grafananotificationchannel/grafananotificationchannel_controller.go

```diff
@@ -0,0 +1,367 @@
+package grafananotificationchannel
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	defaultErrors "errors"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	grafanav1alpha1 "github.com/integr8ly/grafana-operator/api/integreatly/v1alpha1"
+	"github.com/integr8ly/grafana-operator/controllers/constants"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/integr8ly/grafana-operator/controllers/common"
+	"github.com/integr8ly/grafana-operator/controllers/config"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	ControllerName = "controller_grafananotificationchannel"
+)
+
+type GrafanaChannel struct {
+	ID                    *uint   `json:"id"`
+	UID                   *string `json:"uid"`
+	Name                  *string `json:"name"`
+	Type                  *string `json:"type"`
+	IsDefault             *bool   `json:"isDefault"`
+	SendReminder          *bool   `json:"sendReminder"`
+	DisableResolveMessage *bool   `json:"disableResolveMessage"`
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
```

Comment:
Unsure which one is never but I think it would nice to be consistent with the current solution.
In the other reconcilers we use `func (r *ReconcileGrafana) SetupWithManager(mgr ctrl.Manager) error {`
For example https://github.com/grafana-operator/grafana-operator/blob/bf225bd89c236495602ddb9dfaa73a7cca1e630a/controllers/grafana/grafana_controller.go#L60